src/Equinor.ProCoSys.IPO.WebApi/Synchronization/BusReceiverService.cs

```diff
@@ -35,118 +62,188 @@ public async Task ProcessMessageAsync(PcsTopic pcsTopic, Message message, Cancel
 
             switch (pcsTopic)
             {
+                case PcsTopic.Ipo:
+                    await ProcessIpoEvent(messageJson);
+                    break;
                 case PcsTopic.Project:
-                    var projectEvent = JsonSerializer.Deserialize<ProjectTopic>(messageJson);
-                    _plantSetter.SetPlant(projectEvent.ProjectSchema);
-                    ProcessProjectEvent(projectEvent);
+                    ProcessProjectEvent(messageJson);
                     break;
-
                 case PcsTopic.CommPkg:
-                    var commPkgEvent = JsonSerializer.Deserialize<CommPkgTopic>(messageJson);
-                    _plantSetter.SetPlant(commPkgEvent.ProjectSchema);
-                    ProcessCommPkgEvent(commPkgEvent);
+                    ProcessCommPkgEvent(messageJson);
                     break;
-
                 case PcsTopic.McPkg:
-                    var mcPkgEvent = JsonSerializer.Deserialize<McPkgTopic>(messageJson);
-                    _plantSetter.SetPlant(mcPkgEvent.ProjectSchema);
-                    ProcessMcPkgEvent(mcPkgEvent);
+                    ProcessMcPkgEvent(messageJson);
                     break;
             }
             await _unitOfWork.SaveChangesAsync(token);
         }
 
-        private void ProcessMcPkgEvent(McPkgTopic mcPkgEvent)
+        private void ProcessMcPkgEvent(string messageJson)
+        {
+            var mcPkgEvent = JsonSerializer.Deserialize<McPkgTopic>(messageJson);
+
+            _telemetryClient.TrackEvent(IpoBusReceiverTelemetryEvent,
+                new Dictionary<string, string>
+                {
+                    {BusReceiverTelemetryConstants.Event, IpoTopic.TopicName},
+                    {BusReceiverTelemetryConstants.McPkgNo, mcPkgEvent.McPkgNo},
+                    {BusReceiverTelemetryConstants.ProjectSchema, mcPkgEvent.ProjectSchema[4..]},
+                    {BusReceiverTelemetryConstants.ProjectName, mcPkgEvent.ProjectName.Replace('$', '_')}
+                });
+            _plantSetter.SetPlant(mcPkgEvent.ProjectSchema);
+            _invitationRepository.UpdateMcPkgOnInvitations(mcPkgEvent.ProjectName, mcPkgEvent.McPkgNo, mcPkgEvent.Description);

```

Comment:
Hmm ... is this the way to do it?
This mean that we set data in ipo database without any validation (for instance exceeding column max size in db will cause exception). 
If I've written this, I would have created a command which had its own validation + handling. The save on line 78 should then been at end of handler code. The same goes for other handling in this class

Reply:
As discussed. For now we leave it like this, as we will always trust our originator of data. Eg. the module owning the mc package is responsible for validating and all other modules should trust the master.